cns/cnsclient/error_test.go

```diff
@@ -0,0 +1,46 @@
+package cnsclient
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/cns/types"
+)
+
+func TestIsNotFound(t *testing.T) {
+	tests := []struct {
+		name string
+		err  error
+		want bool
+	}{
+		{
+			name: "is not found",
+			err: &CNSClientError{
+				Code: types.UnknownContainerID,
+				Err:  fmt.Errorf("unknown container id"),
+			},
+			want: true,
+		},
+		{
+			name: "is not cnsclienterr",
+			err:  fmt.Errorf("error"),
+			want: false,
+		},
+		{
+			name: "is other cnsclienterr",
+			err: &CNSClientError{
+				Code: types.UnexpectedError,
+				Err:  fmt.Errorf("unexpected err"),
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
If you are running tests in parallel, you shouldn't close over the loop variable

Reply:
should close over the loop variable, will fix. 
@ramiro-gamarra ty, this led me to discover [this](https://github.com/Azure/azure-container-networking/pull/956)